src/components/environment-variables/environment-variables-list.js

```diff
@@ -42,127 +45,109 @@ const EnvironmentVariablesList = (props) => {
     setHasNonRadixEnvVars(checkHasNonRadixEnvVars(edEnvVars));
     setEditableEnvVars(edEnvVars);
   }, [includeRadixVars, inEditMode, envVars]);
+
   const handleSetEditMode = () => {
     setPoolingState({ paused: true });
     setInEditMode(true);
   };
+
   const handleSave = () => {
     if (readonly) {
       return;
     }
+
     const updatableEnvVars = getUpdatableEnvVars(editableEnvVars);
     if (updatableEnvVars.length > 0) {
       saveFunc(updatableEnvVars);
     }
     setInEditMode(false);
     setPoolingState({ paused: false });
   };
+
   const handleReset = () => {
     resetState();
     setInEditMode(false);
     setPoolingState({ paused: false });
   };
+
   return (
-    <React.Fragment>
+    <>
       <div className="section__heading_with_buttons grid grid--gap-medium">
         <Typography variant="h4">Environment variables</Typography>
-        {editableEnvVars &&
-          editableEnvVars.length > 0 &&
-          hasNonRadixEnvVars &&
+        {editableEnvVars?.length > 0 &&
           !readonly &&
-          !inEditMode &&
-          (saveState.status === requestStates.IDLE ||
-            saveState.status === requestStates.SUCCESS) && (
-            <Button
-              onClick={() => {
-                handleSetEditMode();
-              }}
-            >
-              <Icon data={edit} />
-              Edit
-            </Button>
-          )}
-        {editableEnvVars &&
-          editableEnvVars.length > 0 &&
-          !readonly &&
-          inEditMode &&
+          hasNonRadixEnvVars &&
           (saveState.status === requestStates.IDLE ||
             saveState.status === requestStates.SUCCESS) && (
-            <div className="horizontal-buttons grid grid--gap-medium">
-              <Button
-                variant="contained"
-                color="primary"
-                onClick={() => {
-                  handleSave();
-                }}
-              >
-                <Icon data={save} />
-                Apply
-              </Button>
-              <Button
-                variant="outlined"
-                color="primary"
-                onClick={() => {
-                  handleReset();
-                }}
-              >
-                <Icon data={restore_page} />
-                Cancel
-              </Button>
-            </div>
+            <>
+              {inEditMode ? (
+                <div className="horizontal-buttons grid grid--gap-medium">
+                  <Button variant="contained" onClick={() => handleSave()}>
+                    <Icon data={save} /> Apply
+                  </Button>
+                  <Button variant="outlined" onClick={() => handleReset()}>
+                    <Icon data={restore_page} /> Cancel
+                  </Button>
+                </div>
+              ) : (
+                <Button onClick={() => handleSetEditMode()}>
+                  <Icon data={edit} /> Edit
+                </Button>
```

Comment:
make header section responsive. Suggestion:
```suggestion
                <div className="horizontal-buttons">
                  <div>
                    <Button variant="contained" onClick={() => handleSave()}>
                      <Icon data={save} /> Apply
                    </Button>
                  </div>
                  <div>
                    <Button variant="outlined" onClick={() => handleReset()}>
                      <Icon data={restore_page} /> Cancel
                    </Button>
                  </div>
                </div>
              ) : (
                <div>
                  <Button onClick={() => handleSetEditMode()}>
                    <Icon data={edit} /> Edit
                  </Button>
                </div>
```

Reply:
This will result in the Save and Cancel buttons appearing too close
![image](https://user-images.githubusercontent.com/32248439/131974710-7586b609-3cd6-44b6-b781-7d9e7b09762b.png)

I could wrap the Edit button in a `<div>` if you'd prefer

Reply:
I added a second comment which apparently did not come through. I suggest some styling which will fix this: 
```
.section__heading_with_buttons {
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(13rem, 1fr));
  align-items: center;
}

.section__heading_with_buttons button > span {
  grid-auto-columns: max-content;
}

.horizontal-buttons {
  display: flex;
  gap: var(--eds_spacing_medium);
}
```

Reply:
added